src/server/pkg/worker/api_server.go

```diff
@@ -490,6 +489,22 @@ func HashDatum(pipelineInfo *pps.PipelineInfo, data []*Input) (string, error) {
 		hash.Write(datum.FileInfo.Hash)
 	}
 
+	hash.Write([]byte(pipelineInfo.Pipeline.Name))
+	hash.Write([]byte(pipelineInfo.Salt))
+
+	return client.HashPipelineName(pipelineInfo.Pipeline.Name) + hex.EncodeToString(hash.Sum(nil)), nil
```

Comment:
Hmm, that's a good point we could not include it in the hash (which we're doing on line 492 above) however we still want to include the `HashPipelineName` prefix because that's used by garbage collection.